### ISSUE TITLE
Console: Raw SQL editor executes statements implicitly in context of `public` schema

### ISSUE THREAD

**Issue by 1u0** [hasura/graphql-engine#3288]
This is rather a UX issue:

from user point of view, if I switch to another (non `public` schema) in `Data` section of console,
the SQL editor misleadingly looks like if it would execute queries under the currently selected schema by default (**although**, the url of the editor is not schema specific).

Currently, when a query is executed in the editor, if the PostgreSQL schema is not explicitly provided, the query would be implicitly executed in `public` schema.
This makes different user experience when writing SQL statements for `public` schema vs other schema.

One proposal is to make the editor to execute queries under the currently selected schema (that visible on the right panel). This would also improve ergonomics to create tables/views/functions via SQL editor without explicitly writing schema for other schemas that are not `public`.

**Comment by rikinsk** [comment]
The purpose of the Raw SQL is to run queries as you would run directly on the database. The current behavior of using 'public' schema if not mentioned is default Postgres behavior and is not altered by the GraphQL engine in any way. Hence it would not be possible (and in fact something we should not do) to execute queries under the selected the schema in the console as suggested.

Reducing the scope of this issue to ensuring that the Raw SQL tab in the console is not confused to be schema specific.

**Comment by 1u0** [comment]
> Reducing the scope of this issue to ensuring that the Raw SQL tab in the console is not confused to be schema specific.

Agree, this may be an alternative way to go. I think in that case, an expected behaviour would be that the editor passes SQL statements as is (should also work when the default schema in the DB is not `public`, but something else).

Just for clarification, what I mean that Raw SQL editor in the console implicitly adds `public` schema.
The editor parses SQL statements and uses `parseCreateSQL` function that has `public` schema hardcoded :
https://github.com/hasura/graphql-engine/blob/637c2cb55525afb303738daa145ac295fc82fdde/console/src/components/Services/Data/RawSQL/utils.js#L33-L41

**Comment by rikinsk** [comment]
Right. I understand the reason for your confusion now. 

The `parseCreateSQL` function is only used for Hasura specific actions that need to be done along with executing the SQL on the DB. For example, when you create a table in postgres you might also want to track the table in Hasura at the same time (using the `Track this` checkbox). The `parseCreateSQL` helps extract the required information from the SQL statement to perform the Hasura track table action. It just mimics how Postgres would interpret the SQL statement.
